apps/catalogue/src/views/ResourceDetailsView.vue

```diff
@@ -155,118 +63,49 @@ export default {
     };
   },
   computed: {
-    row() {
+    resourceData() {
       return this.data[0];
     },
-  },
-  methods: {
-    openRouterLink(event, name) {
-      this.$router.push({
-        name: name + "-details",
-        params: event[Object.keys(event)[0]],
-      });
-    },
-    toggleNA() {
-      this.hideNA = !this.hideNA;
-    },
-    toSentence(str) {
-      return (
-        str
-          // insert a space before all caps
-          .replace(/([A-Z])/g, " $1")
-          // uppercase the first character
-          .replace(/^./, function (str) {
-            return str.toUpperCase();
-          })
-      );
-    },
-    renderValue(row, col) {
-      if (row[col.name] === undefined) {
-        return [];
-      }
-      if (
-        col.columnType == "REF_ARRAY" ||
-        col.columnType == "MREF" ||
-        col.columnType == "REFBACK" ||
-        col.columnType == "ONTOLOGY_ARRAY"
-      ) {
-        return row[col.name].map((v) => {
-          if (col.refLabel) {
-            return this.applyJsTemplate(col.refLabel, v);
+    sections() {
+      const comparePosition = (a, b) => a.position < b.position;
+      const isHeading = (meta) => meta.columnType === "HEADING";
+      const isNonSystemField = (meta) => !meta.name.startsWith("mg_");
+
+      return this.tableMetadata.columns
+        .filter(isNonSystemField)
+        .sort(comparePosition)
+        .reduce((accum, item) => {
+          // nest fields (data columns) withing sections (headings)
+          if (isHeading(item)) {
+            accum.push({ meta: item, fields: [] });
           } else {
-            return this.flattenObject(v);
+            if (!accum.length) {
+              accum.push({ meta: item, fields: [] });
+            }
+            accum.at(-1).fields.push({
+              meta: { ...item, primaryTableKey: this.primaryTableKey },
+              value: this.resourceData[item.name],
```

Comment:
after pull request #672 this should change to item.id ?

Reply:
yes , the resourceData is a map created from a 'row' and the 'name' is used as key , if the key changes to id , this should be updated